### DISCUSSION TITLE
Ordering with layout.text and table.text

### DISCUSSION THREAD

**Discussion by yflam1** [deepdoctection/deepdoctection#163] [Q&A] [answered]
Is there a way to make proper orders involving both normal layouts and table? For example, I have a PDF like this:

```
text
text
title
text
list
text
table
text
```

Would it be possible to put table in the correct position (unlike something like

```python
result = "\n".join([layout.text for layout in dp.layouts]) + "\n"
for table in dp.tables:
    result += table.text
```

) using your `TextOrderService`? I would like to put everything into one single `.txt` file.

**Comment by JaMe76** [comment, marked as answer]
I am currently refactoring text ordering and text presentation in [no_hierachy_reading_order](https://github.com/deepdoctection/deepdoctection/tree/no_hierachy_reading_order), so that your question can be affirmatived answered. 
I hope that I can merge this branch by the end of this week. 
Currently, it is not easily possible.

**Reply by yflam1** [reply]
Thanks! Looking forward to seeing that implementation!
